### DISCUSSION TITLE
Method to return bounding boxes/polyons?

### DISCUSSION THREAD

**Discussion by mattyli** [geopandas/geopandas#3269] [Ideas]
I know that geopandas already offers the `.bounds` method that returns a dataframe of the bounds for each geometry, I was wondering whether it would be useful or not to implement a method that returns a `GeoSeries` of bounding boxes (polygons) for the specified geometry. I use this function in my project and think that others may find this useful. 

I use the same method used here to generate the corners of the bounding polygons as in this PR [#2197](https://github.com/geopandas/geopandas/pull/2197/files/6c3c1c8099694de4d9f9591c33e99b25613b152a), full credits to the author there.

This is my first discussion post so apologies if I got some of the formatting and stuff wrong. Love the work that you guys are doing here!

```python
import numpy as np
from shapely import polygons 
import geopandas as gpd
from typing import Union

def create_bounding_boxes(input_geometry: Union[gpd.GeoSeries, gpd.GeoDataFrame])->gpd.GeoSeries:
    # Do your basic checks

    bounds = input_geometry.bounds

    # Use numpy to create the vertices of each respective bounding box
    bottom_left_vertex = np.array((bounds.minx.values, bounds.miny.values)).T.reshape(-1, 1, 2)
    bottom_right_vertex = np.array((bounds.maxx.values, bounds.miny.values)).T.reshape(-1, 1, 2)
    top_left_vertex = np.array((bounds.minx.values, bounds.maxy.values)).T.reshape(-1, 1, 2)
    top_right_vertex = np.array((bounds.maxx.values, bounds.maxy.values)).T.reshape(-1, 1, 2)

    bbox_vertices = np.concatenate((bottom_left_vertex,
                                    bottom_right_vertex,
                                    top_right_vertex,
                                    top_left_vertex
                                    ),
                                   axis=1
                                   )
    
    return gpd.GeoSeries(data = polygons(bbox_vertices), index=input_geometry.index, crs=input_geometry.crs)
```

![nybb](https://github.com/geopandas/geopandas/assets/49624796/9250b7d3-ef3a-4921-847a-e1b807705d11)
![borough breakdown of bounding box](https://github.com/geopandas/geopandas/assets/49624796/4efe0f7b-0ef8-4a93-8d6d-401c7799ad1a)

**Comment by theroggy** [comment]
Have you checked out [gdf.envelope](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoSeries.envelope.html)?

**Reply by mattyli** [reply]
oh my no I have not, thanks for pointing that out! hahaha
